src/rgw/driver/d4n/d4n_directory.h

```diff
@@ -64,7 +64,7 @@ class ObjectDirectory: public Directory {
     int get(const DoutPrefixProvider* dpp, CacheObj* object, optional_yield y);
     int copy(const DoutPrefixProvider* dpp, CacheObj* object, const std::string& copyName, const std::string& copyBucketName, optional_yield y);
     int del(const DoutPrefixProvider* dpp, CacheObj* object, optional_yield y);
-    int update_field(const DoutPrefixProvider* dpp, CacheObj* object, std::string field, std::string value, optional_yield y);
+    int update_field(const DoutPrefixProvider* dpp, CacheObj* object, const std::string& field, std::string& value, optional_yield y);
```

Comment:
did you check whether this compiles fine, it gives me errors

Reply:
What is the error? It compiles on my end so that is strange

Reply:
there is a follow up commit that fixes the error

Reply:
Apologies, I meant to squash all commits related to this change into one.